src/Client/index.ts

```diff
@@ -18,6 +18,10 @@ import { debug } from "../utils";
 import { discoverEndpoint } from "./discovery";
 import { parseConnectionString } from "./parseConnectionString";
 
+interface ClientSettings {
```

Comment:
**SUGGESTION:** I think that we should be consistent and name either options or settings (preferable as it's used as such in .NET) for of those interfaces.